### PR TITLE
Add missing SocketAddress API docs

### PR DESCRIPTION
**Pull request by antonfirsov**
Contributes to https://github.com/dotnet/runtime/issues/88569.

### CODE REVIEW

**Review by liveans — APPROVED** [review]

File: xml/System.Net/SocketAddress.xml [hunk 1]
```diff
@@ -217,9 +217,9 @@
         <ReturnType>System.Memory&lt;System.Byte&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Represents underlying memory that can be passed to native OS calls.</summary>
         <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <remarks>The content of the buffer can be invalidated if <see cref="Size"/> is changed or if the SocketAddress is used in another receive call.</remarks>
```
Comment (carlossanlop) [inline xml/System.Net/SocketAddress.xml:222]: There is a CI warning due to an incomplete docID:

```suggestion
        <remarks>The content of the buffer can be invalidated if <see cref="P:System.Net.SocketAddress.Size"/> is changed or if the <see cref="T:System.Net.SocketAddress"/> instance is used in another receive call.</remarks>

```

File: xml/System.Net/SocketAddress.xml [hunk 2]
```diff
@@ -217,9 +217,9 @@
         <ReturnType>System.Memory&lt;System.Byte&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Represents underlying memory that can be passed to native OS calls.</summary>
```
Comment (carlossanlop) [inline xml/System.Net/SocketAddress.xml:220]:
```suggestion
        <summary>Represents the underlying memory that can be passed to native OS calls.</summary>

```

File: xml/System.Net/SocketAddress.xml [hunk 3]
```diff
@@ -446,9 +446,9 @@
         <Parameter Name="addressFamily" Type="System.Net.Sockets.AddressFamily" Index="0" FrameworkAlternate="net-8.0" />
       </Parameters>
       <Docs>
-        <param name="addressFamily">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="addressFamily">The <see cref="T:System.Net.Sockets.AddressFamily"/>.</param>
```
Comment (carlossanlop) [inline xml/System.Net/SocketAddress.xml:449]: It feels ambiguous. Maybe we could use the [official description of the AddressFamily type](https://github.com/dotnet/dotnet-api-docs/blob/9573c5bd5dae4ab6df6de20f89c7d370a743a9dc/xml/System.Net.Sockets/AddressFamily.xml#L45):

```suggestion
        <param name="addressFamily">The addressing scheme to use.</param>

```

**Review by carlossanlop — APPROVED** [review]
LGTM modulo some suggestions for you to consider and a CI warning fix.

File: xml/System.Net/SocketAddress.xml (hunk 2 above)
Comment (gewarren) [reply xml/System.Net/SocketAddress.xml:220]:
```suggestion
        <summary>Gets the underlying memory that can be passed to native OS calls.</summary>

```

**Review by gewarren — APPROVED** [review]

**Pull request merged**